tests/unit/dom.ts

```diff
@@ -488,5 +488,36 @@ registerSuite({
 			assert.strictEqual(parent.children.length, 0);
 			assert.isNull(node.parentNode);
 		}
+	},
+
+	create: {
+		'correct element created'() {
+			let tagType = 'div';
+			let result = dom.create(tagType);
+			assert.equal(result.nodeName.toLowerCase(), tagType);
+		},
+
+		'attributes are set on element'() {
+			let args: CreateArgs = { foo: 'bar', baz: 'buz', attributes: { 'class': 'test' } },
+				element = dom.create('div', args);
+
+			assert.strictEqual((<any> element).foo, 'bar');
+			assert.strictEqual((<any> element).baz, 'buz');
+			assert.strictEqual(element.className, 'test');
+		},
+
+		'String children are appended to element'() {
+			let element = dom.create('div', null, ['<div id="test"></div>']);
```

Comment:
This is not supposed to be allowed by this API.  The point is to allow a convenience for adding text nodes as children (and the purpose of doing that is usually speciflcally to *avoid* injecting HTML).